crates/sdl-encoder/src/field.rs

```diff
@@ -0,0 +1,241 @@
+use crate::{FieldValue, InputValue};
+use std::fmt::{self, Display};
+/// The __Field type represents each field in an Object or Interface type.
+///
+/// *FieldDefinition*:
+///     Description<sub>opt</sub> Name ArgumentsDefinition<sub>opt</sub> **:** TypeDirectives<sub>\[Const\] opt</sub>
+///
+/// Detailed documentation can be found in [GraphQL spec](https://spec.graphql.org/draft/#sec-The-__Field-Type).
+///
+/// ### Example
+/// ```rust
+/// use sdl_encoder::{FieldValue, Field, InputValue};
+///
+/// let ty_1 = FieldValue::Type {
+///     ty: "CatBreed".to_string(),
+/// };
+///
+/// let mut field = Field::new("cat".to_string(), ty_1);
+///
+/// let value_1 = FieldValue::Type {
+///     ty: "CatBreed".to_string(),
+/// };
+///
+/// let arg = InputValue::new("breed".to_string(), value_1);
+///
+/// field.arg(arg);
+/// field.default(Some("\"Norwegian Forest\"".to_string()));
+///
+/// assert_eq!(
+///     field.to_string(),
+///     r#"  cat(breed: CatBreed): CatBreed = "Norwegian Forest""#
+/// );
+/// ```
+#[derive(Debug, PartialEq, Clone)]
+pub struct Field {
+    // Name must return a String.
+    name: String,
+    // Description may return a String.
+    description: Option<String>,
+    // Args returns a List of __InputValue representing the arguments this field accepts.
+    args: Vec<InputValue>,
+    // Type must return a __Type that represents the type of value returned by this field.
+    type_: FieldValue,
+    // Deprecated returns true if this field should no longer be used, otherwise false.
+    is_deprecated: bool,
+    // Deprecation reason optionally provides a reason why this field is deprecated.
+    deprecation_reason: Option<String>,
+    // Default is used to set a default value for fields **only** for fields in Input Objects.
+    default: Option<String>,
```

Comment:
This shouldn't be necessary if we can use `InputValue` or `InputField` for fields of input objects.